### ISSUE TITLE
non-ascii IMAP folder names not properly encoded

### ISSUE THREAD

**Issue by kparal** [pulb/mailnag#88]
This is in my mailnag.cfg:
```
[account1]
enabled = 1
name = user.name (Gmail)
user = user.name
server = imap.gmail.com
port = 993
ssl = 1
imap = 1
idle = 0
folder = INBOX,Gar&AOEBfg-,Issues,Upozorn&ARs-n&AO0-,V&AP0-pisy
```
Please notice the folder names (they display the same in GUI configuration). Instead, they should read:
```
INBOX
Garáž
Issues
Upozornění
Výpisy
```
(because those are the names of my gmail folders). But if I specified them in GUI, mailnag didn't work. I had to encode them by hand. It would be much easier for users if mailnag did that automatically.

IMAP folder names encoding can be done from python. Unfortunately the code is not present in the standard library:
http://stackoverflow.com/questions/12776679/imap-folder-path-encoding-imap-utf-7-for-python
http://bugs.python.org/issue5305
But the code can be copied from permissively licensed sources:
http://imapclient.freshfoo.com/browser/imapclient/imap_utf7.py
https://github.com/barosl/imap-dups/blob/master/imapUTF7.py
http://www.php2python.com/wiki/function.imap-utf7-encode/

**Comment by cheshire-mouse** [comment]
I made this patch for my own needs. Patch fixes the problem in rather primitive way, but it works for me
[nonasciifolders.patch.txt](https://github.com/pulb/mailnag/files/405616/nonasciifolders.patch.txt)

**Comment by pulb** [comment]
@cheshire-mouse: If you've got a clean and GPL-compatible solution for this problem I'd be very interested in a pull request :smile:

**Comment by pulb** [comment]
Fixed in commit 44d250e3214e5358ae50ad45bce32ea1bb52f82b. Thanks to @cheshire-mouse!
